src/ui/shared/openUrl.ts

```diff
@@ -0,0 +1,31 @@
+import browser from 'webextension-polyfill';
+import { setUrlContext } from './setUrlContext';
+import type { UrlContext } from './UrlContext';
+
+async function getNextToActiveTabIndex() {
+  const [activeTab] = await browser.tabs.query({
+    active: true,
+    lastFocusedWindow: true,
+  });
+  return (activeTab?.index ?? -1) + 1;
```

Comment:
I just realized that `0` index is a bad default. We shouldn't open the tab as the first one, but as a last one. So it's better to introduce null here I think

Reply:
I just can't imagine how should we handle that `null` value. If `null` represents the case when there is no active tab then we should create the first tab anyway and it will get `index = 0`, right?

Reply:
If no active tab is found, we should open tab as usual, making it last

Reply:
i.e. not pass `index` argument at all

Reply:
What if I return `Promise<number | undefined>`?
I mean:
```
getNextToActiveTabIndex(): Promise<number | undefined>
```
...
```
const index = await getNextToActiveTabIndex();
browser.tabs.create({ url: url.toString(), index });
```
^ thats how I updated this part

Reply:
This is how I updated this part ^